src/main/java/org/spongepowered/api/text/dialogue/DialogueArchetype.java

```diff
@@ -0,0 +1,198 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.dialogue;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataQuery;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.text.channel.MessageChannel;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Collection;
+
+/**
+ * Represents a structure for a back-and-forth communication between one or
+ * more {@link Speaker}s and one or more plugiins.
+ */
+public interface DialogueArchetype {
+
+    /**
+     * Creates a new {@link Builder} to build a new {@link DialogueArchetype}.
+     *
+     * @return The newly created builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Creates a new {@link Dialogue} for the specified {@link Speaker}(s).
+     *
+     * @param speakers The speakers to send the initial {@link Question} to
+     * @return The newly created {@link Dialogue}
+     * @throws IllegalArgumentException If {@code speakers} is empty
+     */
+    Dialogue start(Speaker... speakers);
+
+    /**
+     * Creates a new {@link Dialogue} for the specified {@link Speaker}(s).
+     *
+     * @param speakers The speakers to send the initial {@link Question} to
+     * @return The newly created {@link Dialogue}
+     * @throws IllegalArgumentException If {@code speakers} is empty
+     */
+    Dialogue start(Iterable<? extends Speaker> speakers);
+
+    /**
+     * Gets the first question that is shown to the {@link Speaker} when the
+     * {@link Dialogue} is instantiated.
+     *
+     * @return The initial question
+     */
+    Question getInitialQuestion();
+
+    /**
+     * Gets the initial data that the generated {@link Dialogue} will be
+     * populated with. This is a copy of the data.
+     *
+     * @return The data
+     */
+    DataContainer getInitialData();
+
+    /**
+     * Gets whether or not output is suppressed by default.
+     *
+     * @return Whether output is suppressed
+     * @see Builder#suppressesOutput(boolean)
+     */
+    boolean suppressesOutput();
+
+    /**
+     * Gets whether or not messages are suppressed by default.
```

Comment:
@ryantheleach If I get your right your are proposing that we remove
* `suppressesOutput()`
* `suppressesAllMessages()`
* `suppressesOutputToPeers()`
* `getAllowedChannels()`

In favor of a new method

* `getOtherChatHandler() : DialogueChatHandler`

**DialogueChatHandler**
````java
public interface DialogueChatHandler {

    /**
     * Processes the given chat details. This method is called after the last chat event 
     * was processed but before the message is actually send to the client.
     *
     * <p>Implementations may queue the given (some) messages to send it after the dialogue is finished.</p>
     * 
     * @param dialogue The dialogue that is currently open.
     * @param details The details of the chat that has been send to a Speaker and has been intercepted.
     * @returns True, if the intercepted message should be forwarded to the client. 
     *             False, if the message was or will be handled by this instance or should be discarded.
     */
    public boolean process(Dialogue dialogue, ChatDetails details);

    /**
     * Notifies this handler of the end of the dialogue.
     * 
     * <p>This method can be used to send queued messages after the dialogue has been completed.</p>
     */
    public void notifyEnd(Dialogue dialogue);

}
````
**ChatDetails**
````java
public interface ChatDetails {

    public Text getMessage();

    public ChatType getType();

    public MessageChannel getChannel();

    [...]

}
````